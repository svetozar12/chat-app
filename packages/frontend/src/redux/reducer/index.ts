import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
