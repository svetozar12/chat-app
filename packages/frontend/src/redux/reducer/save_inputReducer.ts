import { Action, ActionType, InitialState3 } from "../state";
import { HYDRATE } from "next-redux-wrapper";
export const initialState = {
  input_username: "",
  input_password: "",
  input_email: "",
  input_gender: "",
  notification_number: 0,
};
const saveInputReducer = (state: InitialState3 = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.SAVE_INPUT_EMAIL:
      return { ...state, input_email: action.payload };
    case ActionType.SAVE_INPUT_GENDER:
      return { ...state, input_gender: action.payload };
    case ActionType.NOTIFICATION_NUMBER:
      return { ...state, notification_number: action.payload };
    default:
      return state;
  }
};

export default saveInputReducer;
