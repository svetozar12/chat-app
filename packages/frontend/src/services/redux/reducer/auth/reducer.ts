import { HYDRATE } from 'next-redux-wrapper';
import { IAuth } from './state';
import { ActionType } from '../../types';

export const initialState: IAuth = {
  remember_me: false,
  loginPrompt: false,
};
const authReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default authReducer;
