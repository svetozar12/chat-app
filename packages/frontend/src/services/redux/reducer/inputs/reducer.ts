import IInputs from './state';
import { ActionType } from '../../types';

export const initialState: IInputs = {
  input_username: '',
  input_password: '',
  input_email: '',
  input_gender: 'other',
  input_file: null,
  // notification_number: 0,
};
const inputReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.SAVE_INPUT_EMAIL:
      return { ...state, input_email: action.payload };
    case ActionType.SAVE_INPUT_GENDER:
      return { ...state, input_gender: action.payload };
    case ActionType.SAVE_INPUT_FILE:
      return { ...state, input_file: action.payload };
    // case ActionType.NOTIFICATION_NUMBER:
    //   return { ...state, notification_number: action.payload };
    default:
      return state;
  }
};

export default inputReducer;
