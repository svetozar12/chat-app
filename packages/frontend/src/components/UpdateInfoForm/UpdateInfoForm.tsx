import React from "react";
import Link from "next/dist/client/link";
import { useSelector, useDispatch } from "react-redux";
import { InitialState3 } from "../../redux/state";
import { css } from "@emotion/css";
import styled from "@emotion/styled";
import Single_avatar from "../Avatar/Single_avatar/Single_avatar";
const Input_file_container = styled.div`
  margin-top: 2rem;
`;

const Input_file = styled.label`
  background: var(--main-black);
  color: var(--main-white);
  border: 1px solid var(--input-border-color);
  transition: 0.3s;
  padding: 0.5rem;
  cursor: pointer;
  &:hover {
    background: var(--main-white);
    transition: 0.3s;
    color: var(--main-black);
  }
`;

const Button = styled.button`
  width: 20%;
  background: yellow;
`;

export const UpdateInfoForm = ({
  url,
  handleSubmit,
  setImage,
  cookieName,
}: {
  url?: string;
  image?: string;
  handleSubmit: (e: any) => Promise<false | undefined>;
  setImage: React.Dispatch<React.SetStateAction<string>>;
  cookieName: string;
}) => {
  const dispatch = useDispatch();
  const state = useSelector(
    (state: { saveInputReducer: InitialState3 }) => state.saveInputReducer,
  );

  return (
    <main
      title="user_settings"
      className={css`
        display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        hieght: 100%;
      `}
    >
      <section
        className={css`
          width: 25%;
          height: 50vh;
          display: flex;
          flex-direction: column;
          align-items: center;
          border-right: 1px solid rgba(0, 0, 0, 0.2);
        `}
      >
        <h1
          className={css`
            font-size: 2vw;
            margin-right: 1rem;
          `}
        >
          Profile
        </h1>
        <Single_avatar cookieName={cookieName} width="6rem" height="6rem" />
      </section>
      <form style={{ width: "75%", height: "50vh", minWidth: "342px" }}>
        <h2 style={{ fontSize: "2vw" }}>Update your profile info</h2>
        <label htmlFor="">Email</label>
        <input
          type="email"
          value={state.input_email}
          onChange={(e) =>
            dispatch({
              type: "SAVE_INPUT_EMAIL",
              payload: e.target.value,
            })
          }
        />
        <div className="input_gender">
          <div>
            <label htmlFor="Male">Male</label>
            <input
              onChange={(e) =>
                dispatch({
                  type: "SAVE_INPUT_GENDER",
                  payload: e.target.value,
                })
              }
              type="radio"
              name="gender"
              id="Male"
              value="Male"
            />
          </div>
          <div>
            <label htmlFor="Female">Female</label>
            <input
              onChange={(e) =>
                dispatch({
                  type: "SAVE_INPUT_GENDER",
                  payload: e.target.value,
                })
              }
              type="radio"
              name="gender"
              id="Female"
              value="Female"
            />
          </div>
          <div>
            <label htmlFor="Others">Others</label>
            <input
              onChange={(e) =>
                dispatch({
                  type: "SAVE_INPUT_GENDER",
                  payload: e.target.value,
                })
              }
              type="radio"
              name="gender"
              id="Others"
              value="Others"
            />
          </div>
        </div>
        <Input_file_container>
          <input
            type="file"
            name="userAvatar"
            placeholder="Select file"
            className="input_file"
            id="file"
            style={{ display: "none" }}
            onChange={(e: any) => setImage(e.target.files[0])}
          />
          <Input_file className="input_file" htmlFor="file">
            Add file
          </Input_file>
        </Input_file_container>
        <Button
          style={{ fontSize: "1vw", textAlign: "center" }}
          onClick={handleSubmit}
          type="submit"
        >
          Save
        </Button>
        <Button
          style={{ fontSize: "1vw", textAlign: "center" }}
          onClick={handleSubmit}
          type="submit"
        >
          Cancel
        </Button>
      </form>
      <Link href={`/${url}`}>
        <a className="link" style={{ color: "var(--main-blue)" }}>
          Go back
        </a>
      </Link>
    </main>
  );
};
