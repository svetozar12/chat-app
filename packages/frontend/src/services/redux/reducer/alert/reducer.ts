import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
