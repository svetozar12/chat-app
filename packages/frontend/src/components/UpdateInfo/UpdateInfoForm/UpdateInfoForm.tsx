import { FormLabel, GridItem, HStack, Input, Radio, RadioGroup, SimpleGrid, useColorModeValue, VStack } from '@chakra-ui/react';
import useThemeColors from 'hooks/useThemeColors';
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { STATE } from 'services/redux/reducer';
import { setInputEmail, setInputFile, setInputGender, setInputPassword, setInputUsername } from 'services/redux/reducer/inputs/actions';
import IInputs from 'services/redux/reducer/inputs/state';
import { toggleQuickLogin } from 'services/redux/reducer/toggles/actions';
import UpdateInfoDetails from './UpdateInfoDetails';

interface IUpdateInfoForm {
  image: string;
  setImage: React.Dispatch<React.SetStateAction<string>>;
  inputs: IInputs;
  toggleQuickLogin: typeof toggleQuickLogin;
  setInputEmail: typeof setInputEmail;
  setInputGender: typeof setInputGender;
  setInputPassword: typeof setInputPassword;
  setInputUsername: typeof setInputUsername;
}

function UpdateInfoForm(props: IUpdateInfoForm) {
  const { image, inputs, toggleQuickLogin, setInputEmail, setInputGender, setInputPassword, setInputUsername } = props;
  const {
    base: {
      default: { color },
    },
  } = useThemeColors();

  const InputsLayout = [
    {
      label: 'Username',
      labelProps: {
        htmlFor: 'username',
        fontWeight: 'bold',
        fontSize: 'xl',
      },
      props: {
        variant: 'FormInput',
        type: 'username',
        value: inputs.input_username,
        id: 'username',
        w: 'full',
        onChange: (e: any) => setInputUsername(e.target.value),
        color,
      },
    },
    {
      label: 'Email',
      labelProps: {
        htmlFor: 'email',
        fontWeight: 'bold',
        fontSize: 'xl',
      },
      props: {
        variant: 'FormInput',
        type: 'email',
        value: inputs.input_email,
        w: 'full',
        id: 'email',
        color,
        onChange: (e: any) => setInputEmail(e.target.value),
      },
    },
    {
      label: '',
      labelProps: {
        htmlFor: 'file',
        fontWeight: 'bold',
        fontSize: 'xl',
      },
      props: {
        variant: 'FormInput',
        type: 'file',
        id: 'file',
        color,
        value: inputs.input_file,
        display: 'none',
        onChange: (e: any) => setInputFile(e.target.value),
      },
    },
  ];

  return (
    <SimpleGrid my={5} gap={5} columns={2} w="full">
      <GridItem w="full" colSpan={{ base: 2, lg: 1 }}>
        {InputsLayout.map((element, index) => {
          const { label, labelProps, props } = element;
          return (
            <VStack mt={1} key={index} align="flex-start" w="full">
              <FormLabel {...labelProps}>{label}</FormLabel>
              <Input {...props} />
            </VStack>
          );
        })}

        <FormLabel fontWeight="bold" fontSize="xl" mt={1}>
          Gender
        </FormLabel>
        <RadioGroup defaultValue="2">
          <HStack spacing={5} direction="row">
            <Radio colorScheme="blue" value="1">
              Male
            </Radio>
            <Radio colorScheme="pink" value="2">
              Female
            </Radio>
          </HStack>
        </RadioGroup>
      </GridItem>

      <GridItem w="full" colSpan={{ base: 2, lg: 1 }}>
        <UpdateInfoDetails image={image} />
      </GridItem>
    </SimpleGrid>
  );
}

const mapStateToProps = (state: STATE) => ({
  inputs: state.inputs,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  toggleQuickLogin: bindActionCreators(toggleQuickLogin, dispatch),
  setInputUsername: bindActionCreators(setInputUsername, dispatch),
  setInputPassword: bindActionCreators(setInputPassword, dispatch),
  setInputEmail: bindActionCreators(setInputEmail, dispatch),
  setInputGender: bindActionCreators(setInputGender, dispatch),
  setInputFile: bindActionCreators(setInputFile, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(UpdateInfoForm);
