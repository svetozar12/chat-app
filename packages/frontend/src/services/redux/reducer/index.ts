import { AnyAction, combineReducers } from 'redux';
import authReducer from 'services/redux/reducer/auth/reducer';
import { ActionType } from '../types';
import { HYDRATE } from 'next-redux-wrapper';
import wsReducer from 'services/redux/reducer/websocket/reducer';
import messageReducer from 'services/redux/reducer/messages/reducer';

import { IAuth } from 'services/redux/reducer/auth/state';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import { IMessage } from 'services/redux/reducer/messages/state';
import IInputs from 'services/redux/reducer/inputs/state';
import inputReducer from 'services/redux/reducer/inputs/reducer';
import toggleReducer from 'services/redux/reducer/toggles/reducer';
import { IToggle } from 'services/redux/reducer/toggles/state';
import IInvite from 'services/redux/reducer/invites/state';
import inviteReducer from 'services/redux/reducer/invites/reducer';
import { IAlert } from 'services/redux/reducer/alert/state';
import alertReducer from 'services/redux/reducer/alert/reducer';

export interface STATE {
  auth: IAuth;
  ws: IWebSocket;
  messages: IMessage;
  inputs: IInputs;
  toggle: IToggle;
  invite: IInvite;
  alert: IAlert;
}

const combiReducers = combineReducers({
  auth: authReducer,
  ws: wsReducer,
  messages: messageReducer,
  inputs: inputReducer,
  toggle: toggleReducer,
  invite: inviteReducer,
  alert: alertReducer,
});

const combReducers = (state: any, action: AnyAction) => {
  if (action.type === HYDRATE)
    return {
      ...state,
      ...action.payload,
    };
  if (action.type === ActionType.SIGN_OUT) {
    return undefined;
  }
  return combiReducers(state, action);
};

export default combReducers;
