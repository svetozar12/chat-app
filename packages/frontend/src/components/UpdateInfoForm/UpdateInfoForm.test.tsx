import { UpdateInfoForm } from "../UpdateInfoForm/UpdateInfoForm";
import { Provider } from "react-redux";
import { initialState as saveInputState } from "../../redux/reducer/save_inputReducer";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
const mockFunc: any = jest.fn();
const mockStore = configureStore([]);
const store = mockStore({
  saveInputReducer: saveInputState,
});
beforeEach(() => {
  component = renderer
    .create(
      <Provider store={store}>
        <UpdateInfoForm
          url="random.png"
          setImage={mockFunc}
          cookieName="greg"
          handleSubmit={mockFunc}
        />
      </Provider>,
    )
    .toJSON();

  container = render(
    <Provider store={store}>
      <UpdateInfoForm
        url="random.png"
        setImage={mockFunc}
        cookieName="greg"
        handleSubmit={mockFunc}
      />
    </Provider>,
  );
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <UpdateInfoForm/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <UpdateInfoForm/>", () => {
    const renderedComponent = container.getByTitle("user_settings");
    expect(renderedComponent).toBeInTheDocument();
  });
});
