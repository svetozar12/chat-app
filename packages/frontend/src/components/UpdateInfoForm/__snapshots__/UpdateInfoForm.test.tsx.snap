// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <UpdateInfoForm/> 1`] = `
<main
  className="css-1ffcsqf"
  title="user_settings"
>
  <section
    className="css-uh9uvh"
  >
    <h1
      className="css-1ecyjot"
    >
      Profile
    </h1>
    <div
      style={
        Object {
          "marginRight": "1rem",
        }
      }
    >
      <svg
        className="css-nkhozk css-boia29"
        data-testid="single_avatar"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "height": "6rem",
            "width": "6rem",
          }
        }
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
    </div>
  </section>
  <form
    style={
      Object {
        "height": "50vh",
        "minWidth": "342px",
        "width": "75%",
      }
    }
  >
    <h2
      style={
        Object {
          "fontSize": "2vw",
        }
      }
    >
      Update your profile info
    </h2>
    <label
      htmlFor=""
    >
      Email
    </label>
    <input
      onChange={[Function]}
      type="email"
      value=""
    />
    <div
      className="input_gender"
    >
      <div>
        <label
          htmlFor="Male"
        >
          Male
        </label>
        <input
          id="Male"
          name="gender"
          onChange={[Function]}
          type="radio"
          value="Male"
        />
      </div>
      <div>
        <label
          htmlFor="Female"
        >
          Female
        </label>
        <input
          id="Female"
          name="gender"
          onChange={[Function]}
          type="radio"
          value="Female"
        />
      </div>
      <div>
        <label
          htmlFor="Others"
        >
          Others
        </label>
        <input
          id="Others"
          name="gender"
          onChange={[Function]}
          type="radio"
          value="Others"
        />
      </div>
    </div>
    <div
      className="css-zpu9mn"
    >
      <input
        className="input_file"
        id="file"
        name="userAvatar"
        onChange={[Function]}
        placeholder="Select file"
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <label
        className="input_file css-7w5spg"
        htmlFor="file"
      >
        Add file
      </label>
    </div>
    <button
      className="css-lnls4q"
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": "1vw",
          "textAlign": "center",
        }
      }
      type="submit"
    >
      Save
    </button>
    <button
      className="css-lnls4q"
      onClick={[MockFunction]}
      style={
        Object {
          "fontSize": "1vw",
          "textAlign": "center",
        }
      }
      type="submit"
    >
      Cancel
    </button>
  </form>
  <a
    className="link"
    href="/random.png"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={
      Object {
        "color": "var(--main-blue)",
      }
    }
  >
    Go back
  </a>
</main>
`;
